gradle/wrapper/gradle-wrapper.properties

```diff
@@ -1,6 +1,5 @@
-#Sat Nov 03 23:12:45 CET 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
```

Comment:
Do we want to build against a newer gradle version? Might be best to keep this as 4.10.2 for now unless there is a good reason to update. 5.5 has some interesting features that might be of use to use but thats not out yet.

Reply:
Lazy Configuration and the new `Jar` task type's `getArchiveFile()` method required Gradle >= 5.1. I just bumped it to the latest version and added tests for older versions so the build itself would get the benefits of being on a newer version while still checking for compatibility as far back as possible.

Reply:
Ok, if upgrading is needed adding a check to ensure a compatible version is being ran should be fine, its only 1 command to update the wrapper. Might be worth holding out until 5.5 as it has possibly useful features and is currently in RC.